src/Fantomas.CoreGlobalTool.Tests/TestHelpers.fs

```diff
@@ -3,15 +3,27 @@ module Fantomas.CoreGlobalTool.Tests.TestHelpers
 open System
 open System.Diagnostics
 open System.IO
+open System.Text
 
-type TemporaryFileCodeSample internal (codeSnippet: string) =
+type TemporaryFileCodeSample internal (codeSnippet: string, ?hasByteOrderMark: bool) =
+    let hasByteOrderMark = defaultArg hasByteOrderMark false
     let filename = Path.Join(Path.GetTempPath(), Guid.NewGuid().ToString() + ".fs")
-    do File.WriteAllText(filename, codeSnippet)
+    do (if hasByteOrderMark
+        then File.WriteAllText(filename, codeSnippet, Encoding.UTF8)
+        else File.WriteAllText(filename, codeSnippet))
 
     member _.Filename: string = filename
     interface IDisposable with
         member this.Dispose(): unit = File.Delete(filename)
 
+type OutputFile internal () =
+    let filename = Path.Join(Path.GetTempPath(), Guid.NewGuid().ToString() + ".fs")
```

Comment:
This is required to write to the filesystem, isn't it, because of `formatCodeToFile`. I guess it would be a biiiig refactor to switch to `IFileSystem` throughout Fantomas (so that this could be done with an in-memory file system like `MockFileSystem`); probably not a job to be done now.

Reply:
Yes, `runFantomasTool` does the real thing, spins up the cli tool in a new process and actually runs as it would from the command line. IFileSystem and MockFileSystem would be nicer but before February we didn't even have any unit tests of the GlobalCoreTool project so small steps I suppose.